src/Form/index.tsx

```diff
@@ -622,38 +632,34 @@ function Form({
     trackEvent('FeatheryStepLoad', newKey, formName);
 
     let stepChanged = false;
-    await runUserCallback(
-      onLoad,
-      () => {
-        const formattedFields = formatAllFormFields(steps, true);
-        const integrationData: IntegrationData = {};
-        if (initState.authId) {
-          integrationData.firebaseAuthId = initState.authId;
-        }
+    await runUserCallback(onLoad, () => {
+      const formattedFields = formatAllFormFields(steps, true);
+      const integrationData: IntegrationData = {};
+      if (initState.authId) {
+        integrationData.firebaseAuthId = initState.authId;
+      }
 
-        return {
-          fields: formattedFields,
-          stepName: newStep.key,
-          previousStepName: activeStep?.key,
-          // @ts-expect-error TS(2531): Object is possibly 'null'.
-          lastStep: steps[newKey].next_conditions.length === 0,
-          numSteps: Object.keys(steps).length,
-          setStep: (stepKey: any) => {
-            stepChanged = changeStep(stepKey, newKey, steps, history);
-          },
-          setFormCompletion: (flag: boolean) =>
-            client.registerEvent({
-              step_key: newStep.key,
-              event: 'load',
-              completed: flag
-            }),
-          setOptions: updateFieldOptions(steps, newStep),
-          firstStepLoaded: first,
-          integrationData
-        };
-      },
-      newStep
-    );
+      return {
```

Comment:
should be able to clean up the redundant props - verify that they are updated properly after step change due to context changes